devdoc/two_dimensional_array_requirements.md

```diff
@@ -0,0 +1,84 @@
+# two_dimensional_array_requirements
+============================
+
+## Overview
+
+`two_dimensional_array` is a module provide to declare a 2D array with variable size of rows and columns. Each rows has the same size to ensure easier access. The first dimentional of array is resizable, it contains the pointer to the second dimention of array. The pointer is intiazlied to NULL and points to the allocated memory address when there's not enough space. When the memory stored in a row is not needed any more, the entire row shall be freed and set back to `NULL`.
+
+## Exposed API
+```c
+typedef struct TWO_D_ARRAY_TAG
+{
+    uint64_t rows;
+    uint64_t cols;
+} TWO_D_ARRAY_HANDLE;
```

Comment:
>TWO_D_ARRAY_HANDLE [](http://example.com/codeflow?start=2&length=18)

Is the definition of the struct required here?
Can be:
typedef struct TWO_D_ARRAY_TAG* TWO_D_ARRAY_HANDLE;

Reply:
added